pinot-core/src/main/java/com/linkedin/pinot/core/realtime/converter/stats/RealtimeColumnStatistics.java

```diff
@@ -67,21 +68,64 @@ public RealtimeColumnStatistics(RealtimeColumnDataSource dataSource, int[] sorte
 
   @Override
   public Object getMinValue() {
+    if (_dictionaryReader == null) {
```

Comment:
SegmentIndexCreationDriverImpl.buildIndexCreationInfo(). I believe this is only for metadata purpose, but that is to be verified and the segment creation driver re-factored appropriately to reflect the correct metadata.

After I added this, I realized that there is more to do to get the realtime segment built with artificial metadata, and I did not want this PR to grow too big. So I submitted it disabling the feature.